pac4j-ldap/src/main/java/org/pac4j/ldap/credentials/authenticator/LdapAuthenticator.java

```diff
@@ -0,0 +1,124 @@
+/*
+  Copyright 2012 - 2015 pac4j organization
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package org.pac4j.ldap.credentials.authenticator;
+
+import org.ldaptive.LdapAttribute;
+import org.ldaptive.LdapEntry;
+import org.ldaptive.LdapException;
+import org.ldaptive.auth.AuthenticationRequest;
+import org.ldaptive.auth.AuthenticationResponse;
+import org.ldaptive.auth.AuthenticationResultCode;
+import org.ldaptive.auth.Authenticator;
+import org.pac4j.core.exception.AccountNotFoundException;
+import org.pac4j.core.exception.BadCredentialsException;
+import org.pac4j.core.exception.TechnicalException;
+import org.pac4j.core.util.CommonHelper;
+import org.pac4j.http.credentials.UsernamePasswordCredentials;
+import org.pac4j.http.credentials.authenticator.UsernamePasswordAuthenticator;
+import org.pac4j.ldap.profile.LdapProfile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Authenticator for LDAP based on the Ldaptive library and its core {@link org.ldaptive.auth.Authenticator} class.
+ * It creates the user profile and stores it in the credentials for the {@link org.pac4j.http.profile.creator.AuthenticatorProfileCreator}.
+ *
+ * @author Jerome Leleu
+ * @since 1.8.0
+ */
+public class LdapAuthenticator implements UsernamePasswordAuthenticator {
+
+    protected final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private Authenticator ldapAuthenticator;
+
+    private String attributes = "";
+
+    public LdapAuthenticator() {
+    }
+
+    public LdapAuthenticator(final Authenticator ldapAuthenticator) {
+        this.ldapAuthenticator = ldapAuthenticator;
+    }
+
+    public LdapAuthenticator(final Authenticator ldapAuthenticator, final String attributes) {
+        this.ldapAuthenticator = ldapAuthenticator;
+        this.attributes = attributes;
+    }
+
+    public void validate(UsernamePasswordCredentials credentials) {
+        CommonHelper.assertNotNull("ldapAuthenticator", ldapAuthenticator);
+        CommonHelper.assertNotNull("attributes", attributes);
+
+        final String username = credentials.getUsername();
+        final String[] ldapAttributes = attributes.split(",");
```

Comment:
Indeed, I assume it was easier to provide a simple String than a list (in a Spring context for example). It can evolve in the future.